nucypher/blockchain/eth/agents.py

```diff
@@ -762,9 +762,12 @@ def swarm(self) -> Iterable[ChecksumAddress]:
 
     def get_stakers_reservoir(self,
                               duration: int,
-                              without: Iterable[ChecksumAddress] = [],
+                              without: Iterable[ChecksumAddress] = None,
```

Comment:
Is there a reason to make this None and then only check the bool value, setting it to a list?

Reply:
We've talked with Kieran about it just recently in https://github.com/nucypher/nucypher/pull/2352#discussion_r543790073

I don't mind mutable defaults, but if our code style prohibits them, so be it (this one was introduced by me too).

Reply:
Although if we're setting the default to `None` I think the type should be changed to `Optional` as well.

Reply:
> Is there a reason to make this None and then only check the bool value, setting it to a list?

Yes - Mutable defaults are considered an industry-wide anti-pattern.  While I have seen some "use cases" (memoization for example) this particular case does in fact intend for an empty list to be created, and does not leverage any usefulness of mutability.  I'd prefer to stick to common pythonic style, battle tested and widely accepted.

Reply:
:heavy_check_mark: